### DISCUSSION TITLE
Installation requires ssh password prompt because of submodule

### DISCUSSION THREAD

**Discussion by TamaMcGlinn** [krischik/vim-ada#18] [Ideas]
When installing using vim-plug, vim-ada requires an ssh key to be unlocked (assuming you use a password), which is annoying and seems unnecessary. Can we remove the wiki submodule? If we really want the wiki to be available inside every install of vim-ada for some reason, we could instead `git subtree add` it to the repo itself (this preserves its git history).

**Comment by krischik** [comment]
DEIN (https://github.com/Shougo/dein.vim) doesn't load submodule on installation:

```
>ls -la Wiki              
insgesamt 0
drwxr-xr-x  2 martin staff  64 28. Okt 08:56 .
drwxr-xr-x 20 martin staff 640 28. Okt 11:19 ..
```

How do you install your vim bundles?

PS: Is this not more a Discussion (https://github.com/krischik/vim-ada/discussions) then an Issue?

**Comment by TamaMcGlinn** [comment]
That's correct, dein does not. Plug does.

```
call plug#begin('~/.vim/plugged')
Plug 'krischik/vim-ada'
call plug#end()
```

**Reply by TamaMcGlinn** [reply]
Pardon me; that was from memory of using dein a long time ago.
It looks like dein should have the same problem, as [it says here](https://github.com/Shougo/dein.vim/issues/141) that dein initializes submodules.

You are probably not seeing it because your private key is already unlocked when you install vim-ada (or not password protected). Or because dein does not have any decent way to remove plugins at all (you have to do it manually).

For vim-plug, you can reproduce by commenting the Plug line, doing `:PlugClean` to remove vim-ada, and then uncommenting and doing `:PlugInstall`. I get a password prompt across the screen because the remote is specified as ssh rather than http.

But really, we don't need the wiki for the plugin to function, so it just shouldn't be there in the first place.

**Reply by krischik** [reply]
@TamaMcGlinn  I changed from git-ssh urls to https urls so no passwords should be asked any more. It worked on a new system I set up yesterday.

I'll still need to learn how `subtree` works to see if that is closer to what I actually want. The Wiki should only download on development clones not the actual installation.

**Reply by TamaMcGlinn** [reply]
subtree is a one-time command that turns them into a single repo; that means you could write one commit that both changes functionality and changes the wiki. I don't see any problem including the wiki in the actual installation, seeing it's size (20K). Will test if the change works, I expect so.

**Reply by TamaMcGlinn** [reply]
That's great; it works like a charm, now it installs in about 3 seconds, just like other plugins when using Plug.
